tests/pipeline_backend_test.py

```diff
@@ -417,6 +423,43 @@ def tearDownClass(cls):
         cls.sc.stop()
 
 
+class TestLazySingleton(unittest.TestCase):
+
+    def test_with_list(self):
+        ls = LazySingleton([10])
+        self.assertEqual(ls._singleton, 10)
+        self.assertIsNone(ls._iterable)
+        # Also check immediate retrieval
+        self.assertEqual(ls.singleton(), 10)
+
+    def test_init_with_valid_iterable_generator(self):
+        data = (x for x in [10])  # lazy generator
+        ls = LazySingleton(data)
+        self.assertIsNone(ls._singleton)  # Should be lazy
+        self.assertIsInstance(ls._iterable, Iterable)
+        self.assertEqual(ls.singleton(), 10)
+        self.assertEqual(ls.singleton(), 10)
+
+    def test_init_with_empty_list_raises_value_error(self):
+        with self.assertRaisesRegex(ValueError, "exactly one element.*found 0"):
+            LazySingleton([])
+
+    def test_init_with_multi_element_list_raises_value_error(self):
+        with self.assertRaisesRegex(ValueError, "exactly one element.*found 3"):
+            LazySingleton([1, 2, 3])
+
+    def test_init_with_non_iterable_raises_type_error(self):
+        with self.assertRaisesRegex(TypeError,
+                                    "must be a list or an Iterable.*got int"):
+            LazySingleton(123)
+        with self.assertRaisesRegex(
```

Comment:
I think one such assert is enough

just want to avoid accumulating test maintenance burden